R/client.R

```diff
@@ -76,12 +76,15 @@ web_client <- R6::R6Class(
     ##' @param resource_type The type of resource to request (either `Cores`
     ##'   or `Nodes`)
     ##'
+    ##' @param deps Optional. A comma delimited string of dide ids that this job
+    ##'   depends on.
+    ##'
     ##' @param resource_count The number of resources to request
     submit = function(path, name, template, cluster = NULL,
-                      resource_type = "Cores", resource_count = 1) {
+                      resource_type = "Cores", resource_count = 1, deps = "") {
```

Comment:
we have to deal with this at some point, but why at the outside of this function? If you took `NULL` or a character vector then `paste(deps, collapse = ",")` would return the expected format for the endpoint, and would be possibly nicer to work with - that might even be better in `client_body_submit` as it's not totally different in spirit to the base64 encoding that happens there

Reply:
Also consider `depends_on` for consistency with below

Reply:
yeah fair, ok have moved close as possible to usage